cty/set/rules.go

```diff
@@ -22,6 +22,11 @@ type Rules interface {
 	// though it is *not* required that two values with the same hash value
 	// be equivalent.
 	Equivalent(interface{}, interface{}) bool
+
+	// Equals returns true if the instance is equal to another Rules instance.
+	// This method is necessary because some implementations may be backed by
+	// structs which cannot be compared with ==.
+	Equals(Rules) bool
```

Comment:
`Equals` is consistent with how a method like this is named elsewhere, but it feels a little confusing in _this_ case because it's alongside both `Equivalent` which is about comparing potential values, rather than the rules object itself.

What do you think about naming this `SameRules` instead, just to make it a bit clearer that we're talking about comparing the rules object itself? :thinking: